### PR TITLE
Monkey patch TCP keepalives into RiakPbcConnection

### PR DESCRIPTION
**Pull request by mattrobenolt**
/cc @mitsuhiko @dcramer @tkaemming

### CODE REVIEW

File: src/sentry/nodestore/riak/__init__.py [hunk 1]
```diff
@@ -8,3 +8,44 @@
...
+        if not s.getsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE):
+            return
+
+        # Enable TCP keepalive
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
+
+        # Make sure we have TCP_KEEPIDLE before setting it.
+        if hasattr(socket, 'TCP_KEEPIDLE'):
+            # overrides value (in seconds) shown by sysctl net.ipv4.tcp_keepalive_time
+            s.setsockopt(socket.SOL_TCP, socket.TCP_KEEPIDLE, 4 * 60)
```
Comment (dcramer) [inline src/sentry/nodestore/riak/__init__.py]: can we just adjust this on the system instead

Comment (mattrobenolt) [inline src/sentry/nodestore/riak/__init__.py]: We can, but not sure if we want to blindly apply to every TCP connection, unless @mitsuhiko thinks that's fine.

Comment (mattrobenolt) [inline src/sentry/nodestore/riak/__init__.py]: Done.

**Comment by mattrobenolt** [comment]
We stopped using protobuf.

**Pull request closed without merging**